site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -0,0 +1,290 @@
+            currentValue.forEach((val) => {
+                formData.append(key, val);
+            })
+        } else {
+            formData.append(key, currentValue);
+        }
+    })
+
+    return formData;
+}
+
+
+const FileUploader = ({ dropzoneParams, maxFileSize, name, currentFiles, handleDelete }) => {
+    return (
+        <Dropzone {...dropzoneParams}>
+            {({ getRootProps, getInputProps }) => {
+                const inpProps = {
+                    ...getInputProps(),
+                    // Display the dropzone input, otherwise it's hidden by default.
+                    style: {
+                        'display': 'block'
+                    },
+                    name: name
+                }
+                return (
+                    <div {...getRootProps()}>
+                        <input {...inpProps} />
+                        <label className="helptext mt-0">
+                            {`Optional. Max attachments size:`} {humanReadableBytes(maxFileSize, false)}
+                        </label>
+                        {currentFiles.length > 0
+                            &&
+                            <FileTable
+                                handleDelete={handleDelete}
+                                filesList={currentFiles}
+                            />
+                        }
+                    </div>
+                )
+            }}
+        </Dropzone>
+    );
+};
+
+FileUploader.propTypes = {
+    dropzoneParams: PropTypes.object.isRequired,
+    maxFileSize: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    currentFiles: PropTypes.array.isRequired
+};
+
+class SupportForm extends Component {
+
+    constructor(props) {
+        super(props);
+        this.FormSchema = Yup.object({
+            name: Yup.string().required(),
+            email: Yup.string().email().required(),
+            category: Yup.string().required(),
+            subject: Yup.string().required(),
+            description: Yup.string().min(
+                this.props.descriptionMinLength,
+                `Field must be at least ${this.props.descriptionMinLength} characters long.`
+            ).max(
+                this.props.descriptionMaxLength,
+                `Field must be at most ${this.props.descriptionMaxLength} characters long.`)
+                .required(),
+            sysInfo: Yup.boolean()
+        });
+
+        this.state = {
+            files: []
+        };
+    }
+
+    onSubmit = async (values, formikBag) => {
+        const { apiEndpoint } = this.props;
+        let response;
+        const formData = formikToFormData(values);
+        try {
+            response = await http.post(apiEndpoint, formData);
+        } catch (error) {
+            let errMessage = error.message;
+            if (response) {
+                errMessage = error.response.data;
+            }
+            
+            // TODO show error notification
```

Comment:
Currently the form does not provide a lot of feedback to the user, both in error and success scenarios.